src/Main.elm

```diff
@@ -1,10 +1,198 @@
 module Main exposing (main)
 
-import Data.Audience
-import Data.AudienceFolder
-import Html exposing (Html)
+import Browser exposing (sandbox)
+import Data.Audience exposing (..)
+import Data.AudienceFolder exposing (..)
+import Dict exposing (Dict)
+import Dict.Extra as DE exposing (fromListDedupe)
+import Html exposing (Html, div, button, text)
+import Html.Events
+import Html.Attributes
+import Json.Decode as JD exposing (decodeString)
+import List as L exposing (map, drop, filter)
+import Maybe as M
 
 
-main : Html msg
-main =
-    Html.text "There will be app soon!"
+type alias Model = {
+    audiences : Dict Int (List Audience)
+    , audienceFolders :  Dict Int (List AudienceFolder)
+    , currentFolder : List Int
+    , audienceFilter : AudienceType }
+
+type Msg =
+    Up
+    | Down Int
+    | Filter AudienceType
+
+main : Program () Model Msg
+main = sandbox {
+    init = {audiences = audiences
+        , audienceFolders = audienceFolders
+        , currentFolder = []
+        , audienceFilter = Authored},
+    view = view,
+    update = update}
+
+view : Model -> Html Msg
+view model =
+    div [] [viewContents model]
+
+update : Msg -> Model -> Model
+update msg model =
+    case msg of
+        Up -> {model | currentFolder = L.drop 1 model.currentFolder}
+        Down dir -> {model | currentFolder = dir::model.currentFolder}
+        Filter at -> {model | audienceFilter = at, currentFolder = []}
+
+audiences : Dict Int (List Audience)
+audiences =
+    audiencesJSON
+    |> JD.decodeString decodeAudiences
+    |> Result.withDefault []
```

Comment:
Acceptable for the purposes of the challenge :+1:
Can you just sketch what would you probably do if you had to `GET` these JSONs in runtime? eg. what would be the types in the `Model`, etc.